File: application/src/main/java/com/study/platform/domain/comment/port/CommentQueryPort.java
package com.study.platform.domain.comment.port;

import com.study.platform.domain.comment.dto.response.CommentResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface CommentQueryPort {

    Page<CommentResponse> findAllByPostId(UUID postId, Pageable pageable);
}

File: application/src/main/kotlin/com/study/platform/domain/comment/port/CommentQueryPort.kt
package com.study.platform.domain.comment.port

import com.study.platform.domain.comment.dto.response.CommentResponse
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.UUID

interface CommentQueryPort {
    fun findAllByPostId(postId: UUID, pageable: Pageable): Page<CommentResponse>
}
